File: d1/py/d1b.py
from functools import reduce
history = set()
matched = False
counter = 0
accumulated = 0

with open('../input.txt', 'r') as f:
    content = f.readlines()
    results = list(map(int,content))
    
    while (not matched):
      cv = results[counter%len(results)]
      accumulated = accumulated + cv
      if (accumulated in history):
        matched = True
        print(f"Matched found in {accumulated} after {counter} runs")
      else: 
        history.add(accumulated)
        counter += 1

File: js/d1/d1b.js
const fs = require('fs');
const util = require("util");
const readFile = util.promisify(fs.readFile); // wrap the callback with a promise

let match = false
let history = new Set();
let counter = 0
let acc = 0

readFile('input.txt').then( output =>{
  content = output.toString();
  let formatted = formatRawText(content);
  
  do {
    let cv = formatted[counter%formatted.length]
    acc = acc + cv
    if (history.has(acc)) {
      console.log(`Found match at: ${acc}`)
      match = true
    } else {
      counter = counter + 1
      history.add(acc)
    }
  } while (!match)
  
})

function formatRawText(string){
  let array = string.split(/\s+/); // carraige returns
  let parsedArray = array.map(item =>{
    return parseInt(item)
  })
  return parsedArray
}